action/protocol/staking/handlers.go

```diff
@@ -649,40 +649,45 @@ func (p *Protocol) handleCandidateRegister(ctx context.Context, act *action.Cand
 	}
 
 	if err := csm.Upsert(c); err != nil {
-		return log, nil, nil, csmErrorToHandleError(owner.String(), err)
+		return log, nil, csmErrorToHandleError(owner.String(), err)
 	}
 
 	// update bucket pool
 	if err := csm.DebitBucketPool(act.Amount(), true); err != nil {
-		return log, nil, nil, &handleError{
+		return log, nil, &handleError{
 			err:           errors.Wrapf(err, "failed to update staking bucket pool %s", err.Error()),
 			failureStatus: iotextypes.ReceiptStatus_ErrWriteAccount,
 		}
 	}
 
 	// update caller balance
 	if err := caller.SubBalance(act.Amount()); err != nil {
-		return log, nil, nil, &handleError{
+		return log, nil, &handleError{
 			err:           errors.Wrapf(err, "failed to update the balance of register %s", actCtx.Caller.String()),
 			failureStatus: iotextypes.ReceiptStatus_ErrNotEnoughBalance,
 		}
 	}
 	// put updated caller's account state to trie
 	if err := accountutil.StoreAccount(csm, actCtx.Caller, caller); err != nil {
-		return log, nil, nil, errors.Wrapf(err, "failed to store account %s", actCtx.Caller.String())
+		return log, nil, errors.Wrapf(err, "failed to store account %s", actCtx.Caller.String())
 	}
 
 	// put registrationFee to reward pool
-	if err := p.depositGas(ctx, csm, registrationFee); err != nil {
-		return log, nil, nil, errors.Wrap(err, "failed to deposit gas")
+	depositLog, err := p.depositGas(ctx, csm, registrationFee)
```

Comment:
I think this is handled already on line 703

Reply:
yes, let me think about a good way to resolve this problem.